packages/components/src/Table/Filters/Filter.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * This class represents a filter for the elements of a table.
+ * A filter is identified by a unique string identifier.
+ * A filter can be activated or deactivated.
+ */
+export default class Filter {
```

Comment:
Please avoid using class hierarchies, use composition, a class for a filter is total overkill, and local state make it hard to test and maintain.

The only bit with value here being `select` instance method